base/deprecated.jl

```diff
@@ -56,41 +56,33 @@ macro deprecate(old,new)
     end
 end
 
+const have_dep_warned = Set()
+
 function depwarn(msg, funcsym)
     opts = JLOptions()
-    if opts.depwarn > 0
-        ln = Int(unsafe_load(cglobal(:jl_lineno, Cint)))
+    if opts.depwarn == 1  # raise a warning
         fn = String(unsafe_load(cglobal(:jl_filename, Ptr{Cchar})))
+        ln = Int(unsafe_load(cglobal(:jl_lineno, Cint)))
+        key = hash(string(fn, ln, msg))
+        (key in have_dep_warned) && return
```

Comment:
Doesn't this still turn off all the warnings for the same depwarn at different locations in the code called from the same toplevel line?